OutOfSchool/OutOfSchool.WebApi.Tests/Services/DraftStorage/DraftStorageServiceTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models.Workshops;
+using OutOfSchool.BusinessLogic.Services.DraftStorage;
+using OutOfSchool.BusinessLogic.Services.DraftStorage.Interfaces;
+using OutOfSchool.Redis;
+
+namespace OutOfSchool.WebApi.Tests.Services.DraftStorage;
+
+[TestFixture]
+public class DraftStorageServiceTests
+{
+    private Mock<IReadWriteCacheService> readWriteCacheServiceMock;
+    private Mock<ILogger<DraftStorageService<WorkshopBaseDto>>> loggerMock;
+    private IDraftStorageService<WorkshopBaseDto> draftStorageService;
+
+    [SetUp]
+    public void SetUp()
+    {
+        loggerMock = new Mock<ILogger<DraftStorageService<WorkshopBaseDto>>>();
+        readWriteCacheServiceMock = new Mock<IReadWriteCacheService>();
+        draftStorageService = new DraftStorageService<WorkshopBaseDto>(readWriteCacheServiceMock.Object, loggerMock.Object);
+    }
+
+    [Test]
+    public async Task RestoreAsync_WhenDraftExistsInCache_ShouldRestoreAppropriatedEntity()
+    {
+        // Arrange
+        var workshopDraft = new WorkshopBaseDto()
```

Comment:
(Fake data) Test data could and should be randomly generated. See other tests using Faker.

Reply:
Ok, done.
Used Bogus.Facker for creating Workshop draft with random properties.